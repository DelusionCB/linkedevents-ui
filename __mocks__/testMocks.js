export function mockCurrentTime(mockTime) {
    const RealDate = Date

    class MockDate extends RealDate {
        constructor(values) {
            super()
            if (values) {
                return new RealDate(values)
            } else {
                return new RealDate(mockTime)
            }
        }

        static now() {
            return (new RealDate(mockTime)).getTime()
        }
    }
    global.Date = MockDate
}

export const resetMockDate = () => {
    global.Date = window.Date
}

/**
 *
 * @param initialState
 * @param props
 * @returns {*}
 */
export function getProps(initialState, props){
    return {...initialState, ...props}
}
