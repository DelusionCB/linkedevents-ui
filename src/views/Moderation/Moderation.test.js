import configureStore from 'redux-mock-store';
import React from 'react';
import thunk from 'redux-thunk';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import {IntlProvider} from 'react-intl';
import {Helmet} from 'react-helmet';

import testReduxIntWrapper from '../../../__mocks__/testReduxIntWrapper';
import {mockCurrentTime, resetMockDate} from '../../../__mocks__/testMocks';
import ConnectedModeration, {Moderation} from './Moderation';

import mapValues from 'lodash/mapValues';
import fiMessages from 'src/i18n/fi.json';

const testMessages = mapValues(fiMessages, (value, key) => value);
const intlProvider = new IntlProvider({locale: 'fi', messages: testMessages}, {});
const {intl} = intlProvider.getChildContext();

const mockStore = configureStore([thunk]);
const initialStore = {
    user: {
        id: 'testuser',
        username: 'fooUser',
        provider: 'helsinki',
    },
    app: {
        flashMsg: null,
        confirmAction: null,
    },
};

const defaultProps = {
    intl,
};

describe('Moderation Snapshot', () => {
    let store;

    beforeEach(() => {
        mockCurrentTime('2018-11-10T12:00:00z');
    });

    afterEach(() => {
        resetMockDate();
    });

    it('should render view by default', () => {
        const componentProps = {
            confirm: jest.fn(),
            routerPush: jest.fn(),
            setFlashMsg: jest.fn(),
            user: {},
        };
        const wrapper = shallow(<Moderation {...componentProps} />, {context: {intl}});
        expect(wrapper).toMatchSnapshot();
    });

    it('should render view correctly', () => {
        store = mockStore(initialStore);
        const componentProps = {
            confirm: jest.fn(),
            routerPush: jest.fn(),
            setFlashMsg: jest.fn(),
            user: {},
        };
        const componentToTest = <ConnectedModeration {...componentProps} />;
        const wrapper = renderer.create(testReduxIntWrapper(store, componentToTest));

        expect(wrapper).toMatchSnapshot();
    });
});
describe('Moderation', () => {
    function getWrapper(props) {
        return shallow(<Moderation {...defaultProps} {...props} />, {context: {intl}});
    }
    test('react-helmet is defined and gets title prop', () => {
        const wrapper = getWrapper().find(Helmet);
        const pageTitle = wrapper.prop('title');
        expect(wrapper).toBeDefined();
        expect(pageTitle).toBe('Linkedevents - Moderoi sisältöjä');
    });
});
