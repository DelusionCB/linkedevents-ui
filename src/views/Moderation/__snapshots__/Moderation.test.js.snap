// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Moderation Snapshot should render view by default 1`] = `
<div
  className="container"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Linkedevents - Moderoi sisältöjä"
  />
  <h1>
    <FormattedMessage
      id="moderation-page"
      values={Object {}}
    />
  </h1>
  <h2>
    <FormattedMessage
      id="moderation-page-draft-heading"
      values={Object {}}
    />
  </h2>
  <InjectIntl(EventTable)
    count={null}
    events={Array []}
    fetchComplete={true}
    handleInvalidRows={[Function]}
    handlePageChange={[Function]}
    handlePageSizeChange={[Function]}
    handleRowSelect={[Function]}
    handleSortChange={[Function]}
    invalidRows={Array []}
    pageSize={10}
    paginationPage={0}
    selectedRows={Array []}
    sortBy="last_modified_time"
    sortDirection="desc"
    tableCaption="table-event-moderation"
    tableColumns={
      Array [
        "checkbox",
        "name",
        "context",
        "publisher",
        "event_time",
        "last_modified_time",
        "validation",
      ]
    }
    tableName="draft"
    user={Object {}}
  />
  <h2>
    <FormattedMessage
      id="moderation-page-published-heading"
      values={Object {}}
    />
  </h2>
  <p>
    <FormattedMessage
      id="moderation-page-published-description"
      values={Object {}}
    />
  </p>
  <InjectIntl(EventTable)
    count={null}
    events={Array []}
    fetchComplete={true}
    handleInvalidRows={[Function]}
    handlePageChange={[Function]}
    handlePageSizeChange={[Function]}
    handleRowSelect={[Function]}
    handleSortChange={[Function]}
    pageSize={10}
    paginationPage={0}
    sortBy="date_published"
    sortDirection="desc"
    tableCaption="table-event-published"
    tableColumns={
      Array [
        "name",
        "context",
        "publisher",
        "event_time",
        "date_published",
      ]
    }
    tableName="published"
    user={Object {}}
  />
</div>
`;

exports[`Moderation Snapshot should render view correctly 1`] = `
<div
  className="container"
>
  <h1>
    <span>
      Moderoi sisältöjä
    </span>
  </h1>
  <h2>
    <span>
      Odottaa hyväksyntää
    </span>
  </h2>
  <strong>
    <span>
      Ei sisältöjä, joita näyttää.
    </span>
  </strong>
  <h2>
    <span>
      Julkaistut sisällöt
    </span>
  </h2>
  <p>
    <span>
      Jos julkaisit tapahtumaehdotuksen valmiin kattotapahtuman alle, sitä ei näytetä tässä näkymässä.
    </span>
  </p>
  <strong>
    <span>
      Ei sisältöjä, joita näyttää.
    </span>
  </strong>
</div>
`;
